clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -160,9 +163,12 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
              shard->memrefs_until_interrupt_ == 0)) {
             report_if_false(
                 shard,
-                (memref.marker.type == TRACE_TYPE_MARKER &&
-                 (memref.marker.marker_type == TRACE_MARKER_TYPE_KERNEL_EVENT ||
-                  memref.marker.marker_type == TRACE_MARKER_TYPE_RSEQ_ABORT)) ||
+                // I-filtered traces don't have all instrs so this doesn't apply.
+                TESTANY(OFFLINE_FILE_TYPE_FILTERED | OFFLINE_FILE_TYPE_IFILTERED,
```

Comment:
I was thinking maybe we should move refactoring-unrelated changes to a separate PR. But that's optional.

Reply:
I was also considering separating the two changes in commit [8d394c9](https://github.com/DynamoRIO/dynamorio/pull/6169/commits/8d394c9de4b159b789fc7f95cbfda719f45d7b24).  One reason not to separate is that there is no simple test that will reproduce: unless the change to C++ for rseq.c were also separated.  (I don't understand why this shows up with the C++ change.)